scripts/Diffraction/isis_powder/routines/InstrumentSettings.py

```diff
@@ -65,9 +65,9 @@ def update_attributes(self, advanced_config=None, basic_config=None, kwargs=None
             self._parse_attributes(self._adv_config_dict, suppress_warnings=suppress_warnings)
         if advanced_config or basic_config:
             self._parse_attributes(self._basic_conf_dict,
-                                   suppress_warnings=(not bool(basic_config or suppress_warnings)))
+                                   suppress_warnings=(not bool(basic_config) or suppress_warnings))
```

Comment:
More of a query than a note, do we have to cast implicitly to `bool`? I think it's done automatically when you put the `not` in front, i.e. there shouldn't be a difference between `not bool(basic_config)` and just `not basic_config`